### PR TITLE
`TimeSeriesScalar` archetype

### LINKED ISSUE #3174: `Scalar` migration to archetypes
- [ ] Python support
  - [ ] Extensions
  - [ ] Unit test (constructors, serialization)
  - [ ] Doc example
- [ ] Rust support
  - [ ] Extensions
  - [ ] Unit test (constructors, rust-to-rust roundtrip)
  - [ ] Doc example
- [ ] C++ support
  - [ ] Extensions
  - [ ] Unit test (constructors, serialization)
  - [ ] Doc example
- [ ] Cross-language roundtrip test
- [ ] Remove old components
  - [ ] Clean up `re_components`
  - [ ] Port viewer to use new types and queries

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Depends on #3396
* Fixes #3174
* Skipped C++ samples since it depends on #3394
* Did not implement roundtrip and unit tests for this one since I figured the examples (which are roundtrip enabled) cover these trivial types well enough

image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3397) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3397)
- [Docs preview](https://rerun.io/preview/8abd42e35d720f58730955dc0e141e091dc0af73/docs) 
- [Examples preview](https://rerun.io/preview/8abd42e35d720f58730955dc0e141e091dc0af73/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

**Comment by Wumpf** [comment]
Draft until #3396 has been landed and integrated

File: crates/re_space_view_time_series/src/view_part_system.rs [hunk 1]
```diff
@@ -140,46 +130,40 @@ impl TimeSeriesSystem {
...
             );
 
-            let components = Self::archetype_array();
-            let ent_views = range_entity_with_primary::<re_components::Scalar, 6>(
-                store, &query, ent_path, components,
-            );
+            let arch_views = range_archetype::<
+                TimeSeriesScalar,
+                { TimeSeriesScalar::NUM_COMPONENTS },
+            >(store, &query, ent_path);
```
Comment (emilk) [inline crates/re_space_view_time_series/src/view_part_system.rs:136]: weird formatting O.o

File: crates/re_space_view_time_series/src/view_part_system.rs [hunk 2]
```diff
@@ -140,46 +130,40 @@ impl TimeSeriesSystem {
...
+                    arch_view.iter_optional_component::<ScalarScattering>()?,
+                    arch_view.iter_optional_component::<Color>()?,
+                    arch_view.iter_optional_component::<Radius>()?,
+                    arch_view.iter_optional_component::<Text>()?,
                 ) {
-                    Ok(_) | Err(QueryError::PrimaryNotFound(_)) => {}
-                    Err(err) => {
-                        re_log::error_once!("Unexpected error querying {ent_path:?}: {err}");
-                    }
+                    // TODO(andreas): Support entity path
```
Comment (emilk) [inline crates/re_space_view_time_series/src/view_part_system.rs:150]: support it how?

File: crates/re_types/src/components/scalar_ext.rs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+use super::Scalar;
+
+impl std::fmt::Display for Scalar {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:.prec$}", self.0, prec = crate::DISPLAY_PRECISION)
```
Comment (emilk) [inline crates/re_types/src/components/scalar_ext.rs:5]: is it wise to have a hard-coded precision like this? we can give the control to the user instead with 

```suggestion
        self.0.fmt(f)
```

**Review by emilk — APPROVED** [review]

File: crates/re_types/src/components/scalar_ext.rs (hunk 3 above)
Comment (Wumpf) [reply crates/re_types/src/components/scalar_ext.rs:5]: I think you're right, now pondering if I should do the same to radius where I copy pasted this from :/

File: crates/re_space_view_time_series/src/view_part_system.rs (hunk 1 above)
Comment (Wumpf) [reply crates/re_space_view_time_series/src/view_part_system.rs:136]: trying, but can't get it to format differently without introducing a type alias

**Pull request merged**